### PR TITLE
Add/listing page/carousel

### PR DESCRIPTION
**Pull request by jacobvenable**
This PR does not directly address an issue.

## What does this PR do?

Adds a `<Carousel/>` component that will eventually be used for the image slider found on #29.

### CODE REVIEW

File: src/components/Carousel.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+const Carousel = props => {
+  return typeof props.images === 'object' && props.images.length ? (
```
Comment (rhinodavid) [inline src/components/Carousel.jsx]: Currently looking at implementing typescript for the project which will help with some of this

File: src/components/Carousel.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    <div>
```
Comment (rhinodavid) [inline src/components/Carousel.jsx]: If we don't actually need a div here (which we should avoid) you can use [`React.Fragment`](https://reactjs.org/docs/fragments.html) (also [see shorthand](https://reactjs.org/docs/fragments.html#short-syntax) -- not sure if we can use it with our build system yet)

File: src/components/Carousel.jsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+      {props.images.map(image => {
```
Comment (rhinodavid) [inline src/components/Carousel.jsx]:
```suggestion
           {props.images.map(image => 
               <img key={image.id} src={image.url} alt={image.description} />
      )}
```

Equivalent but feels so good

Comment (rhinodavid) [inline src/components/Carousel.jsx]:
```
      {props.images.map(image => 
         <img key={image.id} src={image.url} alt={image.description} />
      )}
```

equivalent but feels so good

File: src/components/Resource.jsx [hunk 4]
```diff
@@ -1,81 +1,96 @@
...
+    }
+  ]
 };
 
-export const ResourceComponent = () => {
-  return <div />;
+const Resource = props => {
+  const { images } = singleResourceApiReturn;
+  return (
+    <div>
```
Comment (rhinodavid) [inline src/components/Resource.jsx]: Same comment about Fragments

**Comment by jacobvenable** [comment]
@rhinodavid other than the TS comment, addressed comments in the latest commit. Thanks!

File: src/components/Resource.jsx [hunk 5]
```diff
@@ -1,81 +1,96 @@
...
+    }
+  ]
 };
 
-export const ResourceComponent = () => {
-  return <div />;
+const Resource = props => {
+  const { images } = singleResourceApiReturn;
+  return (
+    <>
```
Comment (rhinodavid) [inline src/components/Resource.jsx]: Actually on second thought I don't think we even need the Fragment here. Should just be able to do return(<Carousel images={images} />)

File: src/components/Carousel.jsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+const Carousel = props => {
+  return typeof props.images === 'object' && props.images.length ? (
+    <>
```
Comment (gordonhgraham) [inline src/components/Carousel.jsx]: typo? new syntax i'm unaware of? looks like a fragment...?

File: src/components/Carousel.test.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+describe('<Carousel/>', () => {
+  let wrapper, props;
+
+  beforeEach(() => {
+    props = {
+      images: [
+        {
+          id: '123yvu1231bv23',
+          url: 'https://placeholder.com/480x200.png',
```
Comment (gordonhgraham) [inline src/components/Carousel.test.js]: may i recommend [Fill Murray](https://www.fillmurray.com/)

Comment (gordonhgraham) [reply src/components/Carousel.test.js]: 😹

File: src/components/Carousel.jsx (hunk 6 above)
Comment (jacobvenable) [reply src/components/Carousel.jsx]: yep, that's just a short version of a fragment:
https://reactjs.org/docs/fragments.html#short-syntax

Comment (rhinodavid) [reply src/components/Carousel.jsx]: see https://reactjs.org/docs/fragments.html#short-syntax

motivation is we don't want to put extra nodes in the dom tree (`<div>`s) if we don't actually need them semantically.

File: src/components/Resource.jsx (hunk 5 above)
Comment (jacobvenable) [reply src/components/Resource.jsx]: eventually we will need a fragment, there's just only one component being returned at the moment. The next PR will most likely add another sibling of `<Carousel>`

File: src/components/Carousel.test.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        }
+      ]
+    };
+    wrapper = shallow(<Carousel {...props} />);
+  });
+  it('renders all images and only once', () => {
+    let renderedImages;
+    props.images.forEach(image => {
+      renderedImages = wrapper
+        .findWhere(child => {
```
Comment (gordonhgraham) [inline src/components/Carousel.test.js]: i'm not crazy about this... rather tough to read. i think it's the chainign of all these methods... `.forEach().findWhere().map()` i'll try to find you an example of another way to do it or Maybe we can pair on another solution.

File: src/components/Carousel.test.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+            child.prop('src') === image.url &&
+            child.prop('alt') === image.description
+          );
+        })
+        .map(renderedImage => renderedImage.prop('src'));
+      expect(renderedImages).toContain(image.url);
+      expect(renderedImages.length).toBe(1);
+    });
+  });
+  it("doesn't render if props are not provided", () => {
```
Comment (gordonhgraham) [inline src/components/Carousel.test.js]: take a look at [prop types](https://reactjs.org/docs/typechecking-with-proptypes.html) That could eliminate two of these tests, checking if props are provided and checking if they're the correctd type.

File: src/components/Carousel.jsx (hunk 6 above)
Comment (gordonhgraham) [reply src/components/Carousel.jsx]: 👏  thanks guys

File: src/components/Resource.jsx (hunk 5 above)
Comment (rhinodavid) [reply src/components/Resource.jsx]: Build it when you need it

File: src/components/Carousel.test.js (hunk 9 above)
Comment (rhinodavid) [reply src/components/Carousel.test.js]: https://github.com/CodeForBoulder/upswyng/pull/47

sell proptypes >> buy typescript

File: src/components/Resource.jsx (hunk 5 above)
Comment (jacobvenable) [reply src/components/Resource.jsx]: You're kill'n me smalls. Updated in last commit.

File: src/components/Carousel.test.js (hunk 7 above)
Comment (jacobvenable) [reply src/components/Carousel.test.js]: 🤣 I had no idea this existed. I'll have to keep that in mind for next placeholders.

File: src/components/Carousel.test.js (hunk 9 above)
Comment (jacobvenable) [reply src/components/Carousel.test.js]: I'll leave this as is until we decide which direction to go. We'll discuss on #47

**Pull request merged**